filebeat/input/filestream/fswatch.go

```diff
@@ -359,6 +362,18 @@ func (s *fileScanner) shouldSkipFile(file string) bool {
 		return true
 	}
 
+	originalFile, err := filepath.EvalSymlinks(file)
+	if err != nil {
+		s.log.Debugf("finding path to original file has failed %s: %+v", file, err)
+		return true
+	}
+	// Check if original file is included to make sure we are not reading from
+	// unwanted files.
+	if s.isFileExcluded(originalFile) || !s.isFileIncluded(originalFile) {
```

Comment:
Nice. Another potential mitigation we might want to implement (via setting) is to check file ownership. That is, the owner of the symlink must be the owner of the original file, such that user X can not create symlinks for files owned by user Y.

Reply:
Great idea! However, I will implement it in a separate PR as the ownership check is platform-dependent, and might get bigger. I would keep this PR as small as possible.